sotodlib/site_pipeline/update_det_match.py

```diff
@@ -429,13 +425,19 @@ def main(config_file: str, all: bool=False):
 
     runner = Runner(cfg)
 
+    remaining_detsets = runner.get_remaining_detsets()
+    logger.info(f"{len(remaining_detsets)} detsets to match")
     if all:
         update_manifests_all(runner)
-        while runner.run_next_match():
+        for detset in remaining_detsets:
+            run_match(runner, detset)
             update_manifests_all(runner)
     else:
-        runner.run_next_match()
-        update_manifests_all(runner)
+        if len(remaining_detsets) == 0:
+            logger.info("No detsets to run.")
+        else:
+            run_match(runner, remaining_detsets[0])
+            update_manifests_all(runner)
```

Comment:
Would it make sense here to keep trying to run a remaining_detset until one actually succeeds?

In the current code, if it gets hung up on a failed one, but can't add it to failed cache, then it will not be able to run anything else in the mean time.

Reply:
Sure, that makes sense to me.